### DISCUSSION TITLE
@Note Annotation request

### DISCUSSION THREAD

**Discussion by gregpaskal** [allure-framework/allure2#2514] [Ideas & Feedback]
I have integrated Allure in my new Java based automation framework and it’s working great. I would like to add a new annotation called @Note that allows an easy way to insert simple documentation and comments in the context of the captured test data. I am willing to perform the development of this feature as I am very comfortable in a variety of languages but I may need some assistance getting start contributing to an open-source project.

I think this feature will likely follow the same design patters that already exist within Allure annotations such as @Step that this implementation might not be too dificult. A few specific requirements I would like for this feature would be the following.

- Unique color (my preference would have been yellow but that is currently being used for Broken). I would like to suggest either a refactor of the broken color to something else or simply choose a color such as organge for note. THe reason for my selection of yellow comes from a usability perspective and the idea that “Post-it” notes typically are yellow and this would be the most intuitive color for the @Note annotation.
- Unique Icon that resembles a sticky (Post-it) note when the @Note feature is utilized. For the same obvious reason above.
- @Note annotation should be easy to implement as there may be a variety of reason a Test or Automation Enginner would want to include simple documentation about a test within a test case. When executing a test step it may be desirable to include input data that was utilized at this stage of the test. It might be advantageous to insert notes when a test step completes with a pass but it has been derived by the automation that the test step was very latent.

I’m hopeful for some feedback and direction on how to move forward with this development or maybe an existing Allure contributor would like to implement it.

Sincerely,
Greg Paskal
CraftOfTesting.com

**Comment by baev** [comment]
@gregpaskal, could you please share some additional context on the request, e.g., an example of how to use such an API and how and where it can be displayed (using MS Paint is okay)?

**Reply by baev** [reply]
I intentionally haven't brought any implementation ideas yet until I better understand the issue and the way you see it should be solved

**Reply by gregpaskal** [reply]
Yes, I can give you a screenshot of how I have found a way to kind of get the functionality I am after. I've also come to realize that the ability to report things such as Notes as well as other statuses is invaluable. For instance, have you ever considered there are more than simple Pass or Fail results when it comes to testing? I'm starting to include new status such as Pass - Warn which means the test technically passed but it encountered some excessive latency along the way.

**Reply by gregpaskal** [reply]
<img width="1855" alt="allure_screen1" src="https://github.com/allure-framework/allure2/assets/16710280/6581726e-6ef1-4eb6-9fc4-c07a5aeb2402">

**Reply by gregpaskal** [reply]
<img width="1066" alt="allure_screen2" src="https://github.com/allure-framework/allure2/assets/16710280/f4a02013-8cfc-442b-ab8b-d1b4a2aca287">

**Reply by gregpaskal** [reply]
What would be helpful is if I could search or filter test results data that includes notes and other annotations. I love the annotation feature and the ability to add my own custom annotations. I am using the capability quite extensively within Allure.
